src/Language/PureScript/TypeChecker/Unify.hs

```diff
@@ -45,6 +47,36 @@ freshType = do
   modify $ \st -> st { checkNextType = t + 1 }
   return $ srcTUnknown t
 
+-- | Remove any ForAlls and ConstrainedType constructors in a type by introducing new unknowns.
+--
+-- This is necessary during type checking to avoid unifying a polymorphic type with a
+-- unification variable.
+instantiatePolyTypeWithUnknowns :: MonadState CheckState m => SourceType -> m SourceType
```

Comment:
I can't think of a case where you'd want to instantiate a constraint like this and not elaborate some value with a corresponding dictionary. That's why there was only one function before.